CMakeLists.txt

```diff
@@ -359,38 +359,46 @@ if (BUILD_CXX_LANGUAGE_PACKAGE)
 
 	link_directories(${LLVM_LIBRARY_DIRS} ${CLANG_LIBRARY_DIRS} ${Boost_LIBRARY_DIRS})
 
-	llvm_map_components_to_libnames(REQ_LLVM_LIBS
-		support core libdriver passes option
-	)
-	foreach(LLVM_TARGET ${LLVM_TARGETS_TO_BUILD})
-		get_property(lib_deps GLOBAL PROPERTY "LLVMBUILD_LIB_DEPS_LLVM${LLVM_TARGET}CodeGen")
-		list(APPEND REQ_LLVM_LIBS "LLVM${LLVM_TARGET}CodeGen")
-		list(APPEND REQ_LLVM_LIBS "${lib_deps}")
-		get_property(lib_deps GLOBAL PROPERTY "LLVMBUILD_LIB_DEPS_LLVM${LLVM_TARGET}AsmParser")
-		if (NOT "${lib_deps}" STREQUAL "")
-			list(APPEND REQ_LLVM_LIBS "LLVM${LLVM_TARGET}AsmParser")
-			list(APPEND REQ_LLVM_LIBS "${lib_deps}")
-		endif()
-	endforeach()
-
-	set(CLANG_LIBRARIES
-		clangASTMatchers
-		clangFrontend
-		clangSerialization
-		clangDriver
-		clangTooling
-		clangParse
-		clangSema
-		clangStaticAnalyzerFrontend
-		clangStaticAnalyzerCheckers
-		clangStaticAnalyzerCore
-		clangAnalysis
-		clangRewriteFrontend
-		clangEdit
-		clangAST
-		clangLex
-		clangBasic
-	)
+    if(LLVM_LINK_LLVM_DYLIB)
+        set(REQ_LLVM_LIBS LLVM)
+    else()
+        llvm_map_components_to_libnames(REQ_LLVM_LIBS
+            support core libdriver passes option
+        )
+        foreach(LLVM_TARGET ${LLVM_TARGETS_TO_BUILD})
+            get_property(lib_deps GLOBAL PROPERTY "LLVMBUILD_LIB_DEPS_LLVM${LLVM_TARGET}CodeGen")
+            list(APPEND REQ_LLVM_LIBS "LLVM${LLVM_TARGET}CodeGen")
+            list(APPEND REQ_LLVM_LIBS "${lib_deps}")
+            get_property(lib_deps GLOBAL PROPERTY "LLVMBUILD_LIB_DEPS_LLVM${LLVM_TARGET}AsmParser")
+            if (NOT "${lib_deps}" STREQUAL "")
+                list(APPEND REQ_LLVM_LIBS "LLVM${LLVM_TARGET}AsmParser")
+                list(APPEND REQ_LLVM_LIBS "${lib_deps}")
+            endif()
+        endforeach()
+    endif()
+
+    if(LLVM_LINK_LLVM_DYLIB)  # Should be CLANG_LINK_CLANG_DYLIB in future LLVM release
```

Comment:
Why did you use 2 if clauses when 1 is sufficient?

Reply:
Because of this commit in clang https://github.com/llvm/llvm-project/commit/05c479491c1d3236cae4eccaa3fcd2794598522c

As noted in the comment, it would be more accurate to use `CLANG_LINK_CLANG_DYLIB` rather than `LLVM_LINK_LLVM_DYLIB`, once it's availbale (i.e. for LLVM-11)

Reply:
Ah, I did not understand that the comment was really only directed at the second if clause.